wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/ProjectServiceApi.java

```diff
@@ -224,10 +225,14 @@ public ProjectConfigDto updateProject(String wsPath, ProjectConfigDto projectCon
       wsPath = absolutize(wsPath);
       projectConfigDto.setPath(wsPath);
     }
-    boolean registeredEarly = projectManager.isRegistered(wsPath);
+
+    RegisteredProject previous = projectManager.getOrNull(wsPath);
+
     RegisteredProject updated = projectManager.update(projectConfigDto);
-    if (!registeredEarly) { // if project config set firstly we will fire event project created
+    if (previous == null) { // if project config set firstly we will fire event project created
```

Comment:
Why can't we use `org.eclipse.che.api.project.server.ProjectManager#isRegistered` here?

Reply:
We have a [PR](https://github.com/eclipse/che/pull/10454) that based on `ProjectManager#isRegistered` and removes this check at all. BTW the PR is blocked by another bug: https://github.com/eclipse/che/issues/11960